From 5f21679af3ad03a1a2a86fd0c2f998fdf9cb6d5d Mon Sep 17 00:00:00 2001
From: Shivani Raval <sraval@habana.ai>
Date: Fri, 04 Apr 2025 10:26:46 +0300
Subject: [PATCH] [JSW-31] module: added jgs files in soc init

Currently added same files as FS1 for JGS for
asic build.

Change-Id: I37b0053464b7cb1713a3d56ee3bf702752e5a1fa
---

diff --git a/include/jgs/hl_pcie.h b/include/jgs/hl_pcie.h
new file mode 100644
index 0000000..3c8a79e
--- /dev/null
+++ b/include/jgs/hl_pcie.h
@@ -0,0 +1,31 @@
+/* SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2024 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+#ifndef NOC_CFG_INCLUDE_JGS_HL_PCIE_H_
+#define NOC_CFG_INCLUDE_JGS_HL_PCIE_H_
+
+#include <zephyr/types.h>
+
+enum bmu_phase {
+	BMU_DEVELOPMENT_MODE,
+	BMU_PRODUCTION_MODE,
+};
+
+enum bmu_sriov_mode {
+	BMU_1PF_MODE,
+	BMU_1PF_1VF_MODE,
+	BMU_1PF_4VF_MODE,
+};
+
+struct bmu_configs {
+	enum bmu_phase phase;
+	enum bmu_sriov_mode sriov;
+	uint32_t hbm_size_mb;	/* depends on hbm type and binning */
+};
+
+int fs1_bmu_init(struct bmu_configs *config);
+
+#endif /* NOC_CFG_INCLUDE_JGS_HL_PCIE_H_ */
diff --git a/src/jgs/hl_pcie.c b/src/jgs/hl_pcie.c
new file mode 100644
index 0000000..f95948e
--- /dev/null
+++ b/src/jgs/hl_pcie.c
@@ -0,0 +1,460 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+ * Copyright 2024 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+#include <jgs_blocks_linux_driver.h>
+
+#include <zephyr/kernel.h>
+#include <hl_pcie.h>
+#include <hl_platform_defs.h>
+#include <jgs_reg_map.h>
+#include <bmu_regs.h>
+#include <pcie_wrap_regs.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(fs1_pcie);
+
+#define BMU_CONFIGS_VERSION	"0.2"
+
+void pcie_wrap_devel_1pf_init(struct bmu_configs *config)
+{
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_HCFG_ADDR_START_PF), 0x40);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_HCFG_ADDR_END_PF), 0x23F);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_START_PF), 0x0);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_END_PF), 0x3F);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_ADDR_MASK_CFG_BAR_PF), 0x3FF);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_ADDR_MASK_MEM_BAR_PF), 0x7FFFF);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_REMAP_GTT_BASE_LSB_0), 0x1FF93);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_REMAP_GTT_BASE_LSB_1), 0x1FFB3);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_REMAP_GTT_BASE_LSB_2), 0x1FFD3);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_REMAP_GTT_BASE_LSB_3), 0x1FFF3);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_REMAP_GTT_BASE_MSB), 0x0);
+}
+
+void pcie_wrap_devel_1vf_init(struct bmu_configs *config)
+{
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_HCFG_ADDR_START_PF), 0x40);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_HCFG_ADDR_END_PF), 0x23F);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_START_PF), 0x0);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_END_PF), 0x3F);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_HCFG_ADDR_START_VF), 0x40);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_HCFG_ADDR_END_VF), 0x23F);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_START_VF), 0x0);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_END_VF), 0x3F);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_ADDR_MASK_CFG_BAR_PF), 0x3FF);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_ADDR_MASK_CFG_BAR_VF), 0x3FF);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_ADDR_MASK_MEM_BAR_PF), 0x7FFFF);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_ADDR_MASK_MEM_BAR_VF), 0x7FFFF);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_REMAP_GTT_BASE_LSB_0), 0x1FF93);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_REMAP_GTT_BASE_LSB_1), 0x1FFB3);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_REMAP_GTT_BASE_LSB_2), 0x1FFD3);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_REMAP_GTT_BASE_LSB_3), 0x1FFF3);
+	CFG_WREG32(IO0_H0_PCIE_WRAP(mmPCIE_WRAP_GTT_ADDR_REMAP_GTT_BASE_MSB), 0x0);
+}
+
+void pcie_wrap_devel_4vf_init(struct bmu_configs *config)
+{
+	LOG_ERR("%s is not implemented", __func__);
+}
+
+void pcie_wrap_prod_1pf_init(struct bmu_configs *config)
+{
+	LOG_ERR("%s is not implemented", __func__);
+}
+
+void pcie_wrap_prod_1vf_init(struct bmu_configs *config)
+{
+	LOG_ERR("%s is not implemented", __func__);
+}
+
+void pcie_wrap_prod_4vf_init(struct bmu_configs *config)
+{
+	LOG_ERR("%s is not implemented", __func__);
+}
+
+void cfg_bmu_devel_1pf_init(struct bmu_configs *config)
+{
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_0), 0x80004000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_1), 0x80006000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_2), 0x80008000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_3), 0x8000A000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_4), 0x8000C000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_5), 0x8000E000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_6), 0x80010000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_7), 0x80012000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_8), 0x80014000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_9), 0x80016000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_10), 0x80018000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_11), 0x8001A000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_12), 0x8001C000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_13), 0x8001E000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_14), 0x80020000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_15), 0x80022000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_0), 0x5FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_1), 0x7FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_2), 0x9FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_3), 0xBFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_4), 0xDFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_5), 0xFFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_6), 0x11FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_7), 0x13FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_8), 0x15FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_9), 0x17FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_10), 0x19FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_11), 0x1BFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_12), 0x1DFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_13), 0x1FFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_14), 0x21FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_15), 0x23FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_0), 0xE0000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_1), 0xE8000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_2), 0xF0000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_3), 0xF8000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_4), 0xE2000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_5), 0xEA000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_6), 0xF2000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_7), 0xFA000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_8), 0xE4000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_9), 0xEC000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_10), 0xF4000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_11), 0xFC000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_12), 0xE6000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_13), 0xEE000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_14), 0xF6000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_15), 0xFE000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_0), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_1), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_2), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_3), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_4), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_5), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_6), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_7), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_8), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_9), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_10), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_11), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_12), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_13), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_14), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_15), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_16), 0x80000000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_16), 0x7FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_16), 0x80000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_16), 0x7FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_17), 0x80001000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_17), 0x17FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_17), 0x90000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_17), 0x7FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_18), 0x80002000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_18), 0x27FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_18), 0xA0000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_18), 0x7FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_19), 0x80003000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_19), 0x37FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_19), 0xB0000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_19), 0x7FF);
+}
+
+void cfg_bmu_devel_1vf_init(struct bmu_configs *config)
+{
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_0), 0x80004000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_1), 0x80006000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_2), 0x80008000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_3), 0x8000A000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_4), 0x8000C000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_5), 0x8000E000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_6), 0x80010000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_7), 0x80012000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_8), 0x80014000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_9), 0x80016000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_10), 0x80018000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_11), 0x8001A000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_12), 0x8001C000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_13), 0x8001E000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_14), 0x80020000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_15), 0x80022000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_0), 0x5FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_1), 0x7FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_2), 0x9FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_3), 0xBFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_4), 0xDFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_5), 0xFFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_6), 0x11FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_7), 0x13FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_8), 0x15FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_9), 0x17FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_10), 0x19FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_11), 0x1BFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_12), 0x1DFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_13), 0x1FFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_14), 0x21FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_15), 0x23FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_0), 0xE0000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_1), 0xE8000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_2), 0xF0000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_3), 0xF8000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_4), 0xE2000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_5), 0xEA000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_6), 0xF2000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_7), 0xFA000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_8), 0xE4000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_9), 0xEC000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_10), 0xF4000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_11), 0xFC000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_12), 0xE6000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_13), 0xEE000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_14), 0xF6000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_15), 0xFE000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_0), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_1), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_2), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_3), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_4), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_5), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_6), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_7), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_8), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_9), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_10), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_11), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_12), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_13), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_14), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_15), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_16), 0x80000000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_16), 0x7FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_16), 0x80000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_16), 0x7FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_17), 0x80001000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_17), 0x17FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_17), 0x90000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_17), 0x7FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_18), 0x80002000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_18), 0x27FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_18), 0xA0000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_18), 0x7FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_START_CGF_19), 0x80003000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_SRC_END_19), 0x37FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_DST_OFFS_19), 0xB0000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_PF_MASK_19), 0x7FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_0), 0x80004000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_1), 0x80006000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_2), 0x80008000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_3), 0x8000A000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_4), 0x8000C000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_5), 0x8000E000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_6), 0x80010000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_7), 0x80012000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_8), 0x80014000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_9), 0x80016000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_10), 0x80018000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_11), 0x8001A000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_12), 0x8001C000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_13), 0x8001E000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_14), 0x80020000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_15), 0x80022000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_0), 0x5FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_1), 0x7FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_2), 0x9FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_3), 0xBFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_4), 0xDFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_5), 0xFFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_6), 0x11FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_7), 0x13FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_8), 0x15FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_9), 0x17FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_10), 0x19FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_11), 0x1BFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_12), 0x1DFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_13), 0x1FFFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_14), 0x21FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_15), 0x23FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_0), 0xE0000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_1), 0xE8000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_2), 0xF0000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_3), 0xF8000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_4), 0xE2000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_5), 0xEA000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_6), 0xF2000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_7), 0xFA000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_8), 0xE4000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_9), 0xEC000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_10), 0xF4000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_11), 0xFC000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_12), 0xE6000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_13), 0xEE000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_14), 0xF6000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_15), 0xFE000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_0), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_1), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_2), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_3), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_4), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_5), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_6), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_7), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_8), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_9), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_10), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_11), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_12), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_13), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_14), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_15), 0x1FFF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_16), 0x80000000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_16), 0x7FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_16), 0x80000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_16), 0x7FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_17), 0x80001000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_17), 0x17FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_17), 0x90000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_17), 0x7FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_18), 0x80002000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_18), 0x27FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_18), 0xA0000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_18), 0x7FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_START_CGF_19), 0x80003000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_SRC_END_19), 0x37FF);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_DST_OFFS_19), 0xB0000);
+	CFG_WREG32(IO0_H0_BMU_CFG(mmBMU_BAR_VF1_MASK_19), 0x7FF);
+}
+
+void cfg_bmu_devel_4vf_init(struct bmu_configs *config)
+{
+	LOG_ERR("%s is not implemented", __func__);
+}
+
+void cfg_bmu_prod_1pf_init(struct bmu_configs *config)
+{
+	LOG_ERR("%s is not implemented", __func__);
+}
+
+void cfg_bmu_prod_1vf_init(struct bmu_configs *config)
+{
+	LOG_ERR("%s is not implemented", __func__);
+}
+
+void cfg_bmu_prod_4vf_init(struct bmu_configs *config)
+{
+	LOG_ERR("%s is not implemented", __func__);
+}
+
+void mem_bmu_1pf_init(struct bmu_configs *config)
+{
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_SRC_START_CGF_0), 0x80000000);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_SRC_END_0), config->hbm_size_mb - 1);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_DST_OFFS_0), 0x0);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_MASK_0), 0x7FFFF);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_SRC_START_CGF_1), 0x800FFC00);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_SRC_END_1), 0xFFFFF);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_DST_OFFS_1), 0xFFC00);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_MASK_1), 0x3FFF);
+}
+
+void mem_bmu_1vf_init(struct bmu_configs *config)
+{
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_SRC_START_CGF_0), 0x80000000);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_SRC_END_0), config->hbm_size_mb - 1);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_DST_OFFS_0), 0x0);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_MASK_0), 0x7FFFF);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_SRC_START_CGF_1), 0x800FFC00);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_SRC_END_1), 0xFFFFF);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_DST_OFFS_1), 0xFFC00);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_PF_MASK_1), 0x3FFF);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_VF1_SRC_START_CGF_0), 0x80000000);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_VF1_SRC_END_0), config->hbm_size_mb - 1);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_VF1_DST_OFFS_0), 0x0);
+	CFG_WREG32(IO0_H0_BMU_MEM(mmBMU_BAR_VF1_MASK_0), 0x3FFFF);
+}
+
+void mem_bmu_4vf_init(struct bmu_configs *config)
+{
+	LOG_ERR("%s is not implemented", __func__);
+}
+
+static int bmu_devel_init(struct bmu_configs *config)
+{
+	switch (config->sriov) {
+	case BMU_1PF_MODE:
+		pcie_wrap_devel_1pf_init(config);
+		cfg_bmu_devel_1pf_init(config);
+		mem_bmu_1pf_init(config);
+		break;
+	case BMU_1PF_1VF_MODE:
+		pcie_wrap_devel_1vf_init(config);
+		cfg_bmu_devel_1vf_init(config);
+		mem_bmu_1vf_init(config);
+		break;
+	case BMU_1PF_4VF_MODE:
+		pcie_wrap_devel_4vf_init(config);
+		cfg_bmu_devel_4vf_init(config);
+		mem_bmu_4vf_init(config);
+		break;
+	default:
+		LOG_ERR("%s unsupported sriov mode %d", __func__, config->sriov);
+		return 1;
+	}
+
+	return 0;
+}
+
+static int bmu_prod_init(struct bmu_configs *config)
+{
+	switch (config->sriov) {
+	case BMU_1PF_MODE:
+		pcie_wrap_prod_1pf_init(config);
+		cfg_bmu_prod_1pf_init(config);
+		mem_bmu_1pf_init(config);
+		break;
+	case BMU_1PF_1VF_MODE:
+		pcie_wrap_prod_1vf_init(config);
+		cfg_bmu_prod_1vf_init(config);
+		mem_bmu_1vf_init(config);
+		break;
+	case BMU_1PF_4VF_MODE:
+		pcie_wrap_prod_4vf_init(config);
+		cfg_bmu_prod_4vf_init(config);
+		mem_bmu_4vf_init(config);
+		break;
+	default:
+		LOG_ERR("%s unsupported sriov mode %d", __func__, config->sriov);
+		return 1;
+	}
+
+	return 0;
+}
+
+/*
+ * BMU configs are taken from H15_BMU_Configs spreadsheet provided by Architecture.
+ * supported bmu config modes:
+ * phase:
+ *   - development
+ *   - production
+ * SRIOV:
+ *   - 1PF
+ *   - 1PF + 1VF
+ *   - 1PF + 4VF
+ */
+int fs1_bmu_init(struct bmu_configs *config)
+{
+	int err;
+
+	LOG_INF("BMU configs version %s", BMU_CONFIGS_VERSION);
+
+	switch (config->phase) {
+	case BMU_DEVELOPMENT_MODE:
+		err = bmu_devel_init(config);
+		break;
+	case BMU_PRODUCTION_MODE:
+		err = bmu_prod_init(config);
+		break;
+	default:
+		LOG_ERR("unsupported BMU mode %d", config->phase);
+		err = -1;
+	}
+
+	return err;
+}
